src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -736,6 +739,33 @@ class RandomDateGenerator : public Generator<bsoncxx::types::b_date> {
     const std::chrono::milliseconds _max;
 };
 
+
+class IncGenerator : public Generator<int64_t> {
+public:
+    IncGenerator(const Node& node, GeneratorArgs generatorArgs)
+        : _actorId{generatorArgs.actorId},
+          _counter{generatorArgs.counter},
```

Comment:
Rather than passing in generatorArgs.counter as a new argument, I would be tempted to take this initializer out, and on line 750 (in the body) have
`_counter = _start + _multiplier*_actorId;`
If you find you don't need multiplier or start otherwise as members, you could move all of that up into the initializer as well. 

Then in evaluate you just need 
`_counter += _step;`